framework/Optimizers/SPSA.py

```diff
@@ -145,7 +145,7 @@ def localStillReady(self, ready, convergence = False):
         if reason == 'just started':
           self.nextActionNeeded = ('start new trajectory',traj)
           break
-        elif reason in ['seeking new opt point']:
+        elif reason in ['seeking new opt point','received recommended point']:
```

Comment:
In the event it helps, you can find a block describing the process/reason status method in the `__init__` definition of the Optimizer on devel.